[CI] FrozenIndexIT testRetryPointInTime failing

Seems like a one-off failure on ARM : https://gradle-enterprise.elastic.co/scans/tests?search.timeZoneId=Europe%2FBucharest&tests.container=org.elasticsearch.index.engine.frozen.FrozenIndexIT&tests.test=testRetryPointInTime

**Build scan:**
https://gradle-enterprise.elastic.co/s/jgxaxl3gwjel2/tests/:x-pack:plugin:frozen-indices:internalClusterTest/org.elasticsearch.index.engine.frozen.FrozenIndexIT/testRetryPointInTime

**Reproduction line:**
`./gradlew ':x-pack:plugin:frozen-indices:internalClusterTest' --tests "org.elasticsearch.index.engine.frozen.FrozenIndexIT.testRetryPointInTime" -Dtests.seed=991387B9FDE04ADC -Dtests.locale=pl-PL -Dtests.timezone=Africa/Algiers -Druntime.java=17`

**Applicable branches:**
main

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.index.engine.frozen.FrozenIndexIT&tests.test=testRetryPointInTime

**Failure excerpt:**
```
java.lang.IllegalStateException: unable to launch a new watch service

  at __randomizedtesting.SeedInfo.seed([991387B9FDE04ADC:EC6786CE761CF5F4]:0)
  at org.elasticsearch.reservedstate.service.FileSettingsService.startWatcher(FileSettingsService.java:201)
  at org.elasticsearch.reservedstate.service.FileSettingsService.setWatching(FileSettingsService.java:142)
  at org.elasticsearch.reservedstate.service.FileSettingsService.startIfMaster(FileSettingsService.java:136)
  at org.elasticsearch.reservedstate.service.FileSettingsService.doStart(FileSettingsService.java:111)
  at org.elasticsearch.common.component.AbstractLifecycleComponent.start(AbstractLifecycleComponent.java:43)
  at org.elasticsearch.node.Node.start(Node.java:1364)
  at org.elasticsearch.test.InternalTestCluster$NodeAndClient.startNode(InternalTestCluster.java:982)
  at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)
  at java.util.concurrent.FutureTask.run(FutureTask.java:264)
  at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:820)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
  at java.lang.Thread.run(Thread.java:833)

  Caused by: java.io.IOException: User limit of inotify watches reached

    at sun.nio.fs.LinuxWatchService$Poller.implRegister(LinuxWatchService.java:264)
    at sun.nio.fs.AbstractPoller.processRequests(AbstractPoller.java:266)
    at sun.nio.fs.LinuxWatchService$Poller.run(LinuxWatchService.java:364)
    at java.lang.Thread.run(Thread.java:833)

```

--- a/server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java
+++ b/server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java
@@ -62,6 +62,7 @@ public class FileSettingsService extends AbstractLifecycleComponent implements C
 
     private volatile FileUpdateState fileUpdateState = null;
     private volatile WatchKey settingsDirWatchKey = null;
+    private volatile WatchKey configDirWatchKey = null;
 
     private volatile boolean active = false;
     private volatile boolean initialState = true;
@@ -206,6 +207,17 @@ public class FileSettingsService extends AbstractLifecycleComponent implements C
         return this.watchService != null;
     }
 
+    private void cleanupWatchKeys() {
+        if (settingsDirWatchKey != null) {
+            settingsDirWatchKey.cancel();
+            settingsDirWatchKey = null;
+        }
+        if (configDirWatchKey != null) {
+            configDirWatchKey.cancel();
+            configDirWatchKey = null;
+        }
+    }
+
     synchronized void startWatcher(ClusterState clusterState, boolean onStartup) {
         if (watching() || active == false) {
             refreshExistingFileStateIfNeeded(clusterState);
@@ -245,10 +257,11 @@ public class FileSettingsService extends AbstractLifecycleComponent implements C
             // We watch the config directory always, even if initially we had an operator directory
             // it can be deleted and created later. The config directory never goes away, we only
             // register it once for watching.
-            enableSettingsWatcher(null, operatorSettingsDir().getParent());
+            configDirWatchKey = enableSettingsWatcher(configDirWatchKey, operatorSettingsDir().getParent());
         } catch (Exception e) {
             if (watchService != null) {
                 try {
+                    cleanupWatchKeys();
                     this.watchService.close();
                 } catch (Exception ignore) {} finally {
                     this.watchService = null;
@@ -322,10 +335,7 @@ public class FileSettingsService extends AbstractLifecycleComponent implements C
         logger.debug("stopping watcher ...");
         if (watching()) {
             try {
-                if (settingsDirWatchKey != null) {
-                    settingsDirWatchKey.cancel();
-                    settingsDirWatchKey = null;
-                }
+                cleanupWatchKeys();
                 fileUpdateState = null;
                 watchService.close();
                 if (watcherThreadLatch != null) {

